### PR TITLE
[Sprint Maintain]Create Unit Test In Change Password Screen

### PR DESCRIPTION
**Pull request by nguyenhuy95dn**
Create Unit Test In Change Password Screen

### CODE REVIEW

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        mPresenter.validateCurrentPasswordInput(currentPass);
+
+        verify(mViewModel, Mockito.never()).onInputCurrentPasswordError(null);
+    }
+
+    @Test
+    public void validateNewPasswordInput() throws Exception {
+        // Give
+        String newPass = "12345";
+        String msg_empty = "Must not empty";
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: Name 'msg_empty' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        String msg_range = "Must range than 6";
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: Name 'msg_range' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    public void getUser() throws Exception {
+        // Give
+        User user = new User();
+        user.setName("Huy");
+        user.setBirthday("11/1/1995");
+        // When
+        when(mUserRepository.getUser()).thenReturn(Observable.just(user));
+        // Then
+        mPresenter.getUser();
+        Mockito.verify(mViewModel, Mockito.never()).onChangePasswordError(null);
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.verify' due to existing static import 'org.mockito.Mockito.verify'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    @Test
+    public void changePassword() throws Exception {
+        // Give
+        Object a = new Object();
+        String currentPass = "123456";
+        String newPass = "12345";
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "12345" appears 7 times in this file; the first occurrence is on line 75
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    @Test
+    public void validateDataInput() throws Exception {
+        // Give
+        boolean expect = true;
+        String currentPass = "123456";
+        String newPass = "12345";
+        String confirmPass = "12345";
+        // When
+        Mockito.when(mValidator.validateAll(mViewModel)).thenReturn(true);
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    @Test
+    public void validateNewPasswordInput() throws Exception {
+        // Give
+        String newPass = "12345";
+        String msg_empty = "Must not empty";
+        String msg_range = "Must range than 6";
+        // When
+        Mockito.when(mValidator.validateValueNonEmpty(newPass)).thenReturn(msg_empty);
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        Mockito.when(mValidator.validateValueRangeMin6(newPass)).thenReturn(msg_range);
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    public void getUser() throws Exception {
+        // Give
+        User user = new User();
+        user.setName("Huy");
+        user.setBirthday("11/1/1995");
+        // When
+        when(mUserRepository.getUser()).thenReturn(Observable.just(user));
+        // Then
+        mPresenter.getUser();
+        verify(mViewModel, Mockito.never()).onChangePasswordError(null);
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: pls verify mViewModel call onChangePasswordSuccess one times

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    @Test
+    public void changePassword() throws Exception {
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: pls write for 2 case success and error

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class ChangePasswordPresenterTest {
+
+    private static final String CURRENT_PASSWORD = "123456";
+    private static final String NEW_PASSWORD = "12345";
+    private static final String CONFIRM_PASSWORD = "12345";
+
+    @InjectMocks
+    ChangePasswordPresenter mPresenter;
+
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: pls remove blank line

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+    ImmediateSchedulerProvider mBaseSchedulerProvider;
+
+    @Before
+    public void setUp() throws Exception {
+        mPresenter.setViewModel(mViewModel);
+        mPresenter.setBaseSchedulerProvider(mBaseSchedulerProvider);
+    }
+
+    @After
+    public void tearDown() throws Exception {
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: pls remove this method cause unused

File: app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+        boolean actual =
+                mPresenter.validateDataInput(CONFIRM_PASSWORD, NEW_PASSWORD, CURRENT_PASSWORD);
+
+        Assert.assertEquals(expect, actual);
+    }
+
+    @Test
+    public void validateCurrentPasswordInput_shouldCorrect_whenInputValidData() throws Exception {
+        // Give
+        String msgNotEmpty = "Must not empty";
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/changepassword/ChangePasswordPresenterTest.java]: WARNING: [AvoidDuplicateLiterals] The String literal "Must not empty" appears 4 times in this file; the first occurrence is on line 143
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Pull request merged**